tests/test_dataset.py

```diff
@@ -60,6 +60,14 @@ def test_select():
     }
 
 
+def test_select_keep_clauses():
+    """Test that calling select doesn't discard filter clauses"""
+    dataset_before = Dataset('test-bucket', ['dim1', 'dim2']).where(dim1=True)
+    dataset_after = dataset_before.select('field1', 'field2')
+
+    assert dataset_before.clauses == dataset_after.clauses
```

Comment:
Could you verify that we keep the other attributes (bucket, schema, store) as well?